quodlibet/quodlibet/qltk/songlistcolumns.py

```diff
@@ -279,14 +279,23 @@ def _apply_value(self, model, iter_, cell, stamp):
             except (OverflowError, ValueError, OSError):
                 text = u""
             else:
-                today = datetime.datetime.now().date()
-                days = (today - date).days
-                if days == 0:
-                    format_ = "%X"
-                elif days < 7:
-                    format_ = "%A"
+                format_setting = config.gettext("settings",
+                                      "datecolumn_timestamp_format")
+
+                # use default behaviour-format
+                if not format_setting:
```

Comment:
Would something like setting it to a format string and running test_qltk_songlistcolumns.py again work? I have no idea how one would test it, other than maybe setting the current time to a specific value and checking whether a date-column formats to what was expected with some entry.